src/extension/provider/annotations.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  window,
+  commands,
+  NotebookCell,
+  NotebookCellOutput,
+  NotebookCellOutputItem,
+  NotebookCellStatusBarItemProvider,
+  NotebookCellStatusBarItem,
+  NotebookCellStatusBarAlignment,
+} from 'vscode'
+
+import { OutputType } from '../../constants'
+import { CellOutputPayload } from '../../types'
+import { Kernel } from '../kernel'
+import { getAnnotations } from '../utils'
+
+export class AnnotationsProvider implements NotebookCellStatusBarItemProvider {
+  constructor(private readonly kernel: Kernel) {
+    commands.registerCommand(
+      'runme.openCellAnnotations',
+      async (cell: NotebookCell) => {
+        try {
+          const exec = await kernel.createCellExecution(cell)
+          exec.start(Date.now())
+          const json = <CellOutputPayload<OutputType.annotations>>{
+            type: OutputType.annotations,
+            output: {
+              annotations: getAnnotations(cell),
+            },
+          }
+          await exec.replaceOutput([
+            new NotebookCellOutput([
+              NotebookCellOutputItem.json(json, OutputType.annotations),
+              NotebookCellOutputItem.json(json),
+            ]),
+          ])
+          exec.end(true)
+        } catch (e: any) {
+          window.showErrorMessage(e.message)
+        }
+      }
+    )
+  }
+
+  async provideCellStatusBarItems(
+    cell: NotebookCell
+  ): Promise<NotebookCellStatusBarItem | undefined> {
+    if (!this.kernel.hasAnnotationsEditExperimentEnabled) {
```

Comment:
Question, if we run this flag check at the extension.ts level before registering the NoteBook provider, we can skip completely checking this flag each time a cell scrolls into view?